### PR TITLE
chore: Add migration for featured shortcuts default widget

### PR DESCRIPTION
**Pull request by abbyoung**
# SC-1717

## Proposed changes

* New migration to add FeaturedShortcuts widget to users' MySpace
* If the user already has the FeaturedShortcuts widget, no action will be taken
* Down migration will only remove FeaturedShortcuts widget added by the migration. Users who added it previously should not be affected.
* To differentiate between users who already have the Featured Shortcuts widget in their MySpace, and users who will receive the update, I added the field `default: true` to the widget.

## Reviewer notes

* Happo diffs seem to be known issue with YouTube thumbnails, but tagging design for review anyway

## Setup

You can test the migration locally in two ways:

### Scenario 1: Log in to the portal app and verify you can see the Featured Shortcuts widget
1. Start the services with `yarn services:up`.  View the mongodb users via `localhost:8888` and verify there is an existing user. If you've recently purged your local db, log in as any user _before_ pulling down the branch. We want to verify that the migration is working correctly, so we need a 'before' state to compare it to.
2. If your user does have the Featured Shortcuts widget, remove it either via UI or Mongo Express.
3. Pull the branch. Run the app, and log in as the same user. You should now see the Featured Shortcuts component in MySpace.

### Scenario 2: Use migrate up/down

1. Pull down the branch. Start services with `yarn services:up`. View the mongodb users via `localhost:8888` and verify data for existing users.
2. If you've already pulled down the branch and logged in (Scenario 1), you'll need to run `yarn migrate down` first, since the migration has already ran. After doing this, the Featured Shortcuts widget should be removed from users that received it from the migration.
3. Run `yarn migrate up`.  Check data in Mongo Express.
  * Any user that did not already have the Featured Shortcuts widget should now have it.
  * If a user already had it, nothing should change.
4. Run `yarn migrate down`.
  * Users with the new default Featured Shortcuts widget should no longer have it.
  * Users who added it manually should still have it.

### As the original developer, I have

- [ ] Met the acceptance criteria, or will meet them in subsequent PRs or stories
  - ... 
- [ ] Created new stories in Storybook if applicable
  - [ ] Checked that all Storybook accessibility checks are passing
- [ ] Created/modified automated unit tests in Jest
  - [ ] Including jest-axe checks when UI changes
- [ ] Created/modified automated E2E tests in Cypress
  - [ ] Including cypress-axe checks when UI changes
  - [ ] Checked that the E2E test build is not failing
- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
  - [ ] Checked responsiveness in mobile, tablet, and desktop
  - [ ] Checked keyboard navigability
  - [ ] Tested with [VoiceOver](https://dequeuniversity.com/screenreaders/voiceover-keyboard-shortcuts) in Safari
  - [ ] Checked VO's [rotor menu](https://github.com/trussworks/accessibility/blob/master/README.md#how-to-use-the-rotor-menu) for landmarks, page heading structure and links
  - [ ] Used a browser a11y tool to check for issues (WAVE, axe, ANDI or Accessibility addon tab for Storybook)
- [ ] Requested a design review for user-facing changes
- For any new migrations/schema changes:
  - [ ] Followed guidelines for zero-downtime deploys

### As code reviewer(s), I have

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
  - [ ] Checked that the E2E test build is not failing
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] Considered marking this as accepted even if there are small changes needed
- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
  - [ ] Checked responsiveness in mobile, tablet, and desktop
  - [ ] Checked keyboard navigability
  - [ ] Tested with [VoiceOver](https://dequeuniversity.com/screenreaders/voiceover-keyboard-shortcuts) in Safari
  - [ ] Checked VO's [rotor menu](https://github.com/trussworks/accessibility/blob/master/README.md#how-to-use-the-rotor-menu) for landmarks, page heading structure and links
  - [ ] Used a browser a11y tool to check for issues (WAVE, axe, ANDI or Accessibility addon tab for Storybook)

### As a designer reviewer, I have

- [ ] Checked in the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow
- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
  - [ ] Checked responsiveness in mobile, tablet, and desktop
  - [ ] Checked keyboard navigability
  - [ ] Tested with [VoiceOver](https://dequeuniversity.com/screenreaders/voiceover-keyboard-shortcuts) in Safari
  - [ ] Checked VO's [rotor menu](https://github.com/trussworks/accessibility/blob/master/README.md#how-to-use-the-rotor-menu) for landmarks, page heading structure and links
  - [ ] Used a browser a11y tool to check for issues (WAVE, axe, ANDI or Accessibility addon tab for Storybook)

### As a test user, I have

- Run through the [Test Script](hhttps://docs.google.com/spreadsheets/d/1eV8UK0aJZ0qrzjnXf5SJ_uRiV7bpsj3yrhRFEkjOvV4/edit?usp=sharing):
  - [ ] On commercial internet in IE11
  - [ ] On commercial internet in Firefox
  - [ ] On commercial internet in Chrome
  - [ ] On commercial internet in Safari
  - [ ] On NIPR in IE11
  - [ ] On NIPR in Firefox
  - [ ] On NIPR in Chrome
  - [ ] On NIPR in Safari
  - [ ] On a mobile device in Firefox
  - [ ] On a mobile device in Chrome
  - [ ] On a mobile device in Safari
- [ ] Added any anomalous behavior to this PR

---

### CODE REVIEW

**Review by shkeating — APPROVED** [review]

File: src/types/index.ts [hunk 1]
```diff
@@ -168,6 +168,8 @@ export type Widget = {
   _id: ObjectId
   title: string
   type: WidgetType
+  // default indicates if the widget is automatically added for the user
+  default?: boolean
```
Comment (gidjin) [inline src/types/index.ts:172]: praise: I like this idea.

**Review by gidjin — APPROVED** [review]
LGTM, left one minor suggestion

File: migrations/1679951734168-add-featured-shortcuts-widget.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
+'use strict'
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const { runMigration } = require('../utils/mongodb')
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const { ObjectId } = require('mongodb')
```
[the remark this answers is not in the document]
Comment (abbyoung) [reply migrations/1679951734168-add-featured-shortcuts-widget.js:6]: strange, i thought i fixed this. there's a 99% chance i didn't push it up 🙃 thank you!

**Review by jcbcapps — APPROVED** [review]
LGTM!

**Pull request merged**